contracts/AcrossEventEmitter.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title AcrossEventEmitter
+ * @notice A simple contract that emits events with bytes encoded metadata
+ */
+contract AcrossEventEmitter is ReentrancyGuard {
+    /**
+     * @notice Emitted when metadata is stored
+     * @param data The metadata bytes emitted
+     */
+    event MetadataEmitted(bytes data);
+
+    /**
+     * @notice Prevents native token from being sent to this contract
+     */
+    receive() external payable {
+        revert("Contract does not accept native token");
+    }
+
+    /**
+     * @notice Prevents native token from being sent to this contract via fallback
+     */
+    fallback() external payable {
+        revert("Contract doesn't accept native token");
+    }
```

Comment:
nit: why implement these functions at all?

Reply:
I considered would be good practice to not allow funds being sent by mistake, but I can remove this